### ISSUE TITLE
Ownership requirements when using .OwnedBy or /createAsOwner to create service principals.

### ISSUE THREAD

**Issue by psignoret** [microsoftgraph/microsoft-graph-docs#16138]
The POST /servicePrincipals articles should be updated to include the ownership requirements when calling with Application.ReadWrite.OwnedBy, as a user who is allowed to create app registrations, or via the .../createAsOwner permission. 

The requirements are:

1. When the calling app is acting on behalf of the signed-in user (i.e. delegated access) and...

   * ...the signed-in user is in a tenant [where registering apps is allowed for all users](https://docs.microsoft.com/en-us/azure/active-directory/roles/delegate-app-roles#restrict-who-can-create-applications), OR, ...
   * ...the signed-in user has the [Application Developer directory role](https://docs.microsoft.com/en-us/azure/active-directory/roles/permissions-reference#application-developer), OR...
   * ...the signed-in user has been assigned a custom directory role with the permission microsoft.directory/servicePrincipals/createAsOwner...

   ...the signed-in user *must* be owner of the **application** object for the **servicePrincipal** being created. 

2. When the calling app is acting as itself (i.e. direct, app-only access) and..

   * ...the calling app has been granted the Microsoft Graph app role (application permission) *Application.ReadWrite.OwnedBy*, OR...
   * ...the calling app's service principal has been assigned the [Application Developer directory role](https://docs.microsoft.com/en-us/azure/active-directory/roles/permissions-reference#application-developer), OR...
   * ...the calling app's service principal has been assigned a custom directory role with the permission microsoft.directory/servicePrincipals/createAsOwner...

   ...the calling app's service principal *must* be owner of the **application** object for the **servicePrincipal** being created. 

In both cases, this also means the application for which a service principal is being created *must* exist in the same tenant where the request is being made.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4b1fa292-64ad-6567-50c9-4219912b0665
* Version Independent ID: 8f228aac-f319-630a-fd9b-b97d51edc1ed
* Content: [Create serviceprincipal - Microsoft Graph v1.0](https://docs.microsoft.com/en-us/graph/api/serviceprincipal-post-serviceprincipals?view=graph-rest-1.0&tabs=http)
* Content Source: [api-reference/v1.0/api/serviceprincipal-post-serviceprincipals.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/main/api-reference/v1.0/api/serviceprincipal-post-serviceprincipals.md)
* Product: **applications**
* Technology: **microsoft-graph**
* GitHub Login: @sureshja
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by jiasli** [comment]
I believe we also need this:

> To be able to create the service principal of an app which is registered in a different tenant, you must be Cloud Application Administrator or Application Administrator.

**Comment by FaithOmbongi** [comment]
Thank you @psignoret, @jiasli for highlighting these doc gaps. 

@psignoret - I've raised a PR to update the document with a note about the requirements in the app-only scenario. Let me get more information about the requirements in the delegated scenario before updating the documentation.

@jiasli  - Confirmed. I've also added this info into the docs PR.

**Comment by psignoret** [comment]
@FaithOmbongi I've re-opened this issue because the current note isn't accurate. Here are all the ways this API does succeed:

When calling on behalf of a signed-in user, with the delegated permission Application.ReadWrite.All, Directory.ReadWrite.All, or Directory.AccessAsUser.All:

| User authorization | For which apps can a service principal be created? | 
|---|--|
| <ul><li>Global Administrator, Application Administrator, or Cloud Application Administrator roles</li><li>A custom directory role with the microsoft.directory/servicePrincipals/create permission</li></ul>| Multi-tenant apps registered in any tenant, and all apps registered in the same tenant |
| <ul><li>A non-admin user in a tenant where app registration by users is allowed</li><li>A user with the Application Developer role</li><li>A user with a custom directory role with the microsoft.directory/servicePrincipals/createAsOwner permission</li></ul>| App registrations the user is owner of, in the same tenant |

When calling directly as the calling app (i.e. app-only request using client credential):

| App authorization | For which apps can a service principal be created? | 
|---|--|
| <ul><li>The application permission Application.ReadWrite.All or Directory.ReadWrite.All</li><li>Global Administrator, Application Administrator, or Cloud Application Administrator roles</li><li>A custom directory role with the microsoft.directory/servicePrincipals/create permission</li></ul>| Multi-tenant apps registered in any tenant, and all apps registered in the same tenant |
| <ul><li>The application permission Application.ReadWrite.OwnedBy</li><li>The Application Developer role</li><li>A user with a custom directory role with the microsoft.directory/servicePrincipals/createAsOwner permission</li></ul>| App registrations the user is owner of, in the same tenant |

**Comment by FaithOmbongi** [comment]
Hi @psignoret, @jiasli - Thak you for your patience.

I've investigated the best way to document this information but unfortunately, the docs platform and Markdown may not be the most user-friendly and best consumable way. So for now, I've added a text after the permissions table that covers delegated scenarios. This is the minimum requirement because delegated scenarios require _both_ permissions and admin roles.

For app-only scenarios, we currently only mention the permissions in the reference docs; then provide more info about the _either permissions or roles_ behavior in the [permissions overview](https://learn.microsoft.com/en-us/graph/permissions-overview?tabs=http#application-permissions).

Soon, we'll publish a dynamic way for developers and apps to identify permissions for calling specific APIs and hopefully, it'll be a better way to consume this information.
